src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9827,7 +9831,11 @@ private void ParseDeclarationModifiers(SyntaxListBuilder list)
                     mod = this.EatToken();
                 }
 
-                if (k is SyntaxKind.ReadOnlyKeyword or SyntaxKind.VolatileKeyword)
+                if (isError)
+                {
+                    mod = this.AddError(mod, ErrorCode.ERR_TypeExpected);
```

Comment:
I don't think this is the right approach. The existing errors around modifiers such as `public` are much better than "Type expected". Let's just adjust the valid modifiers for `using` locals to restrict the modifiers we want, rather than adversely affecting all other modifiers that are already reported.

Reply:
I debated leaving the previously reported modifiers as-is or re-using their diagnostic, but the reason I went with "type expected" instead of the old error ("The modifier '...' is not valid for this item") is that the grammar does not permit any modifiers in `using` declaration.
The old message implies that some modifiers are valid but this one isn't. That message makes sense for a local declaration, but for a `using` declaration it is misleading. In fact, we are only expecting a type at that location.

Reply:
I don't agree with that conclusion. A type was provided, and we parse it as a type. Replacing the provided modifier with a type instead would not make compilation pass, we'd just have an extra identifier in the code now. We'll also report multiple type expected errors in a row now.

Reply:
> Replacing the provided modifier with a type [...]

Why would someone do that? Saying that a type is expected doesn't imply that you should replace the modifier with a type. It just means that a type should follow the `using`.

I'm open to adding a dedicated error message: "Modifiers cannot be placed on resource declarations". That'll match what we do for other similar situations ("Modifiers cannot be placed on event accessor declarations").

Reply:
I would be fine with a dedicated message.